### PR TITLE
Appstore screenshot: Simulate in-person payment dialog for screenshot

### PR DESCRIPTION
**Pull request by itsmeichigo**
⚠️ Please make sure that #7216 is reviewed and merged first: ⚠️  

### Description

This PR continues the work from #7216 to capture the collect payment dialog for app store screenshot. Changes include:

- Hardware: we need to create a mock card reader and card reader status, so public initializers for these had to be added.
- UI Layer: additional accessibility identifiers to access the buttons and detect the collect payment modal.
- Woo Foundation: a minor update to `CurrencyFormatter` was added to transform value from non-Latin to Latin. This is necessary to format the order total value from Arabic numerals to Latin ones. `CollectOrderPaymentUseCase` needs to check this value for collecting payments.
- Yosmite: add mocks for `OrderCardPresentPaymentEligibilityAction`, `SystemStatusAction` and `CardPresentPaymentAction`.

### Testing instructions

- Turn off the feature flag `.splitViewInOrdersTab` in `DefaultFeatureFlagService`.
- Optionally set the simulator to a non-English language to make sure the test isn’t relying on anything English-specific. Try Arabic to check the case with Arabic numerals.
- Select target WooCommerceScreenshots. Run the `testScreenshots()` UI test.
- Confirm that the test succeeds.

You can find the screenshots in the path `~/Library/Caches/tools.fastlane/screenshots`. It's likely that the folder `screenshots` hasn't been created yet, so you might need to create one yourself.

💡 A tip for changing the test language to Arabic: update the language in `~/Library/Caches/tools.fastlane/language.txt` to `ar`.

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/Classes/ViewModels/Order Details/OrderDetailsDataSource.swift
Comment (jaclync) [inline WooCommerce/Classes/ViewModels/Order Details/OrderDetailsDataSource.swift]: super nit: might be nice to be consistent with other accessibility identifiers in order details
```suggestion
                       accessibilityIdentifier: "order-details-collect-payment-button") { [weak self] in
```

File: WooCommerce/Classes/ViewRelated/Orders/Payment Methods/PaymentMethodsView.swift
Comment (jaclync) [inline WooCommerce/Classes/ViewRelated/Orders/Payment Methods/PaymentMethodsView.swift]: nit: can this be a `let` without a default value?

File: WooCommerce/UITestsFoundation/Screens/Orders/CardPresentPaymentsModalScreen.swift
Comment (jaclync) [inline WooCommerce/UITestsFoundation/Screens/Orders/CardPresentPaymentsModalScreen.swift]: super nit:

```suggestion
    /// Button to dismiss the modal
```

File: WooCommerce/UITestsFoundation/Screens/Orders/PaymentMethodsScreen.swift
Comment (jaclync) [inline WooCommerce/UITestsFoundation/Screens/Orders/PaymentMethodsScreen.swift]: super nit: 
```suggestion
    /// Button to dismiss the modal
```

File: WooCommerce/UITestsFoundation/Screens/Orders/PaymentMethodsScreen.swift
Comment (jaclync) [inline WooCommerce/UITestsFoundation/Screens/Orders/PaymentMethodsScreen.swift]: super nit from my understanding of the button:
```suggestion
    /// Button to select card payment method
```

File: WooCommerce/WooCommerceTests/Tools/CurrencyFormatterTests.swift
Comment (jaclync) [inline WooCommerce/WooCommerceTests/Tools/CurrencyFormatterTests.swift]: thanks for adding a test for this ❤️ 

super nit: might be nice to use the new naming convention for test cases:
```suggestion
    func test_arabic_numerals_return_decimal() {
```

would be great to have the Given/When/Then structure in the test case as well!

File: Yosemite/Yosemite/Model/Mocks/MockObjectGraph.swift
Comment (jaclync) [inline Yosemite/Yosemite/Model/Mocks/MockObjectGraph.swift:114]: I was thinking about using Fakeable+Copiable here, but realized that this isn't in a test target 😬

**Review by jaclync — COMMENTED** [review]
Thanks for making all the changes to take this screenshot @itsmeichigo!

Code LGTM, but the test got stuck on `WooCommerceScreenshots.tapCollectPaymentButton()` on this screen:



I did some debugging, and it looks like `SystemStatusAction.synchronizeSystemPlugins` (where we insert the mock plugins) is never called before accessing the WCPay plugin here:

https://github.com/woocommerce/woocommerce-ios/blob/92a92f981ab24d658126ad664926d9b663ef069f/WooCommerce/Classes/ViewRelated/Dashboard/Settings/In-Person%20Payments/CardPresentPaymentsOnboardingUseCase.swift#L176

Did the test use to pass for you? I wonder if something changed like after a merge from the latest `trunk` 🤔

File: WooCommerce/Classes/ViewRelated/Orders/Payment Methods/PaymentMethodsView.swift
Comment (itsmeichigo) [reply WooCommerce/Classes/ViewRelated/Orders/Payment Methods/PaymentMethodsView.swift]: Updated in 233f3cc74bba9eb53ff1242c796309f07e27832f.

File: Yosemite/Yosemite/Model/Mocks/MockObjectGraph.swift
Comment (itsmeichigo) [reply Yosemite/Yosemite/Model/Mocks/MockObjectGraph.swift:114]: Yeah that's why we have such large changes for this file 😅

**Comment by itsmeichigo** [comment]
Thank you @jaclync for the review! I think there were some changes to the payment flow causing this to fail - I had to update the script to sync the necessary data upon starting the app in c826371aa8ab5afc7babccd4e05ac16502fc4a9e. This PR is ready for another look!

File: Yosemite/Yosemite/Model/Mocks/MockStoresManager.swift
Comment (itsmeichigo) [inline Yosemite/Yosemite/Model/Mocks/MockStoresManager.swift:214]: I had to explicitly declare the type of `continuation`, otherwise, the compiler will complain that it cannot infer the type. Please feel free to suggest another solution if possible, I'm still learning to use async-await so I could be wrong somewhere.

Comment (jaclync) [reply Yosemite/Yosemite/Model/Mocks/MockStoresManager.swift:214]: yea I think this is fine, it's probably because there's no return value for the async function so the compiler can't determine the return type.

**Review by jaclync — APPROVED** [review]
Thanks for the updates @itsmeichigo! The test passes now 🎉 

I was looking at the IPP screenshot, just curious why a currency symbol seems to be missing?

**Comment by itsmeichigo** [comment]
Good catch noticing the missing currency! I asked about that in a Slack thread p1657293733486829-slack-C025A8VV728, I think this is a regression from a new update.

**Review by jaclync — APPROVED** [review]
Thanks for checking! Feel free to merge this, while we wait for a review on the formatted amount fix!

**Pull request merged**
